### PR TITLE
[ricos] add container width optional prop

### PR DESCRIPTION
**Pull request by oded-soffrin**
In this PR:
- passing container width prop (need for performance, and experimental truncate-based preview/readmore)
- using in image viewer under experiment - better loading time
- story book example - https://rich-content-storybook-container-width.surge.sh/?path=/story/test-cases--image-plugin-with-width

### CODE REVIEW

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 1]
```diff
@@ -30,7 +31,23 @@ class ImageViewer extends React.Component {
     this.imageRef = React.createRef();
   }
 
+  static contextType = GlobalContext;
+
+  skipImageThumbnail = () => {
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:36]: Opinion: I like it when a method does what it's name means. 
In this case the name says that it skips the thumbnail.
I would name it shouldSkip...

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 2]
```diff
@@ -30,7 +31,23 @@ class ImageViewer extends React.Component {
...
+    } else {
+      return false;
+    }
+  };
+
   componentDidMount() {
+    if (this.skipImageThumbnail()) {
+      return;
+    }
+
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:50]: If the code below is related only to thumbnail then we can put it in the `if`.  
A bit weird to "stop" an unrelated function flow like this.

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 3]
```diff
@@ -74,6 +91,23 @@ class ImageViewer extends React.Component {
...
     };
 
+    const getImageDimensions = (width, isMobile) => {
+      let requiredHeight;
+      let requiredWidth = width || 1;
+      if (isMobile) {
+        //adjust the image width to viewport scaling and device pixel ratio
+        requiredWidth *= (!isSSR() && window.devicePixelRatio) || 1;
+        requiredWidth *= (!isSSR() && window.screen.width / document.body.clientWidth) || 1;
+      }
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:101]:
```suggestion
      if (isMobile && !isSSR()) {
        //adjust the image width to viewport scaling and device pixel ratio
        requiredWidth *= window.devicePixelRatio;
        requiredWidth *= window.screen.width / document.body.clientWidth;
      }
```

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 4]
```diff
@@ -82,25 +116,38 @@ class ImageViewer extends React.Component {
...
+        const {
+          componentData: {
+            config: { size },
+          },
+        } = this.props;
+
+        let effectiveWidth = this.context.containerWidth;
+
+        if (size === 'small') {
+          //350px
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:129]: what's the comment?

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 5]
```diff
@@ -82,25 +116,38 @@ class ImageViewer extends React.Component {
...
+          effectiveWidth = Math.min(effectiveWidth, 350);
         }
-        //keep the image's original ratio
-        requiredHeight = this.calculateHeight(requiredWidth, src);
-        requiredWidth = Math.ceil(requiredWidth);
-        requiredHeight = Math.ceil(requiredHeight);
+        const [w, h] = getImageDimensions(effectiveWidth, this.props.isMobile);
+        requiredWidth = w;
+        requiredHeight = h;
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:134]:
```suggestion
       [requiredWidth, requiredHeight] = getImageDimensions(effectiveWidth, this.props.isMobile);
```

File: packages/plugin-image/web/src/image-viewer.jsx [hunk 6]
```diff
@@ -109,30 +156,17 @@ class ImageViewer extends React.Component {
...
-    if (src && src.fallback) {
-      this.setState({
-        fallbackImageSrc: {
-          preload: src.fallback,
-          highres: src.fallback,
-        },
-      });
-    }
-  };
+  onImageLoadError = () => {};
```
Comment (shaulgo) [inline packages/plugin-image/web/src/image-viewer.jsx:159]: I have no idea what's this. Don't know if we need it. But if we don't, can't we delete the function completely?

**Review by shaulgo — APPROVED** [review]
Only personal styling. Can take any or non of the comments.

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 4 above)
Comment (oded-soffrin) [reply packages/plugin-image/web/src/image-viewer.jsx:129]: currently when size is small it's 350 but can be overrided by consumers, it's WIP under experiment i'll improve the comment

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 5 above)
Comment (oded-soffrin) [reply packages/plugin-image/web/src/image-viewer.jsx:134]: Like! I didn't know we can do that!

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 3 above)
Comment (oded-soffrin) [reply packages/plugin-image/web/src/image-viewer.jsx:101]: not related to my changes, but sure why not

File: packages/plugin-image/web/src/image-viewer.jsx (hunk 6 above)
Comment (oded-soffrin) [reply packages/plugin-image/web/src/image-viewer.jsx:159]: removed somehow by accident, i'll restore

**Comment by oded-soffrin** [comment]
> Only personal styling. Can take any or non of the comments.

Thanks, took care of them all

**Pull request merged**